src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineParser.java

```diff
@@ -138,16 +148,38 @@ private String getUsagePreamble() {
         return usagePreamble;
     }
 
-
+    /**
+     * @param callerArguments The object containing the command line arguments to be populated by
+     *                        this command line parser.
+     */
     public CommandLineParser(final Object callerArguments) {
+        this(callerArguments, null);
+    }
+
+    /**
+     * @param callerArguments The object containing the command line arguments to be populated by
+     *                        this command line parser.
+     * @param pluginDescriptors A list of {@link GATKCommandLinePluginDescriptor} objects that
+     *                          should be used by this command line parser to extend the list of
+     *                          command line arguments with dynamically discovered plugins. If
+     *                          null, no descriptors are loaded.
+     */
+    public CommandLineParser(
+            final Object callerArguments,
+            final List<? extends GATKCommandLinePluginDescriptor<?>> pluginDescriptors) {
         this.callerArguments = callerArguments;
 
-        createArgumentDefinitions(callerArguments);
+        createArgumentDefinitions(callerArguments, null);
+        if (pluginDescriptors != null) {
```

Comment:
I think we should generally prefer passing an empty list vs a null for something like this.